test/metabase/test/util.clj

```diff
@@ -794,3 +796,15 @@
   []
   (with-open [socket (ServerSocket. 0)]
     (.getLocalPort socket)))
+
+(defmacro with-env-keys-renamed-by
+  "Evaluates body with the current core.environ/env being redefined, its keys having been renamed by the given
+  rename-fn."
+  {:arglists '([rename-fn & body])}
+  [rename-fn & body]
+  `(let [e#      env/env
```

Comment:
1.  You should add some `testing` context here. It's free and it it's always nice to have more information than less when debugging failing tests. See the impl of `with-temporary-setting-values` for example. I'd do a `data/diff` on `env/env` before and after the call to `map-keys` and use that.
2. You'd make your life a lot easier if you implemented the actual logic here a function and made `with-env-keys-renamed-by` just a thin wrapper around the function. The impl of `discard-setting-changes` is a good example of what I mean.

   Not only is it a lot easier to understand, but it results in smaller code/faster compilation since we're not repeating the same code everywhere the macro is used

Reply:
In order to capture the full set of renames, to set the testing context, I found it simpler to just build up a map of them (via `reduce-kv`) then use `set/rename-keys` with that.  Also, have moved the bulk of the logic to a fn instead as per your suggestion.